pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -106,14 +107,18 @@ public static boolean containsFunction(String functionName) {
   }
 
   /**
-   * Returns the {@link FunctionInfo} associated with the given function name and number of parameters, or {@code null}
-   * if there is no matching method. This method should be called after the FunctionRegistry is initialized and all
-   * methods are already registered.
+   * Returns a pair <{@code true}, {@link FunctionInfo}> associated with the given function name and number of
+   * parameters if it exists, <{@code true}, {@code null}> if method exists but number of parameters do not match,
+   * else if there is no matching method <{@code false}, {@code null}>. This method should be called after the
+   * FunctionRegistry is initialized and all methods are already registered.
    */
   @Nullable
-  public static FunctionInfo getFunctionInfo(String functionName, int numParameters) {
+  public static ImmutablePair<Boolean, FunctionInfo> getFunctionInfo(String functionName, int numParameters) {
```

Comment:
I feel returning a pair makes this method hard to use. Suggest throwing exception when this method cannot find a matching function registered, which contains the information on whether the function is not registered, or the parameters not match

Reply:
Thank you for your feedback! While trying to refactor with an exception I realised we have `containsFunction` method for `FunctionRegistry` class which can be used for the required functionailty. I will update PR with required logic, apologies for the overlook.

Reply:
Yes, I also noticed that, but that might involve small overhead of canonicalizing function name and map lookup (should be fine as that only happen when this method returns `null`). IMO both way work, and you may choose the one you prefer

Reply:
I felt that using `containsFunction` might improve readability and changes are as in commit https://github.com/apache/pinot/commit/8865d2e14d8dc6de4e84ddcfca7aae9e29dc5364 .

But when it comes to overhead, it might be more than only when it returns `null` due to additional pre-checks in certain cases. Specifically in `InbuiltFunctionEvaluator.planExecution` and constructor of `PostAggregationFunction`. 

Could you comment on performance implications of additional map lookup in these places? If it can be significant, I will resort to creating a new exception.

Reply:
I don't think the overhead is significant, but it is good to eliminate it if possible. See the suggested change